src/solver/impls/imex-bdf2/imex-bdf2.cxx

```diff
@@ -788,18 +806,16 @@ int IMEXBDF2::run() {
 	//Check if we will go past the target time (i.e. past the output step).
 	//If so we want to limit the timestep.
 	//There's potential for this to confuse the adaptive calculation so
-	//we'll set a flag to alert us to this forced change. Not currently used
-	//but we may want to use this to override the dtNext at the end of the 
-	//step to be what we originally wanted to use (i.e. what dtNext is prior
-	//to following if block).
+	//we'll set a flag to alert us to this forced change. 
 	bool artificalLimit = false;
+        BoutReal dtNoLimit = dtNext; // What dt would have been without artificial limit
 	if(cumulativeTime+dtNext > out_timestep){
 	  artificalLimit = true;
 	  dtNext = out_timestep - cumulativeTime;
 	}
 
-	// output << "Attempting internal step "<<counter<<" (attempt "<<adaptCounter<<")"<<endl;
-	// output << "Using dt = "<<dtNext<<endl;
+	//output << "At t=" << cumulativeTime << " attempting internal step "<<counter<<" (attempt "<<adaptCounter<<")"<<endl;
+	//output << "Using dt = "<<dtNext<<endl;
```

Comment:
Can these two lines be deleted? Or uncommented and wrapped in a conditional of some nature?